include/mitsuba/render/shape.h

```diff
@@ -242,6 +246,11 @@ class MTS_EXPORT_RENDER Shape : public Object {
                                                             bool shading_frame = true,
                                                             Mask active        = true) const;
 
+    virtual Float eval_vertex_attribute_1(const std::string& /*name*/, const SurfaceInteraction3f &/*si*/, Mask /*active*/ = true) const { return zero<Float>(); }
+    virtual Color3f eval_vertex_attribute_3(const std::string& /*name*/, const SurfaceInteraction3f &/*si*/, Mask /*active*/ = true) const { return zero<Color3f>(); }
+    virtual Float eval_face_attribute_1(const std::string& /*name*/, const SurfaceInteraction3f &/*si*/, Mask /*active*/ = true) const { return zero<Float>(); }
+    virtual Color3f eval_face_attribute_3(const std::string& /*name*/, const SurfaceInteraction3f &/*si*/, Mask /*active*/ = true) const { return zero<Color3f>(); }
```

Comment:
The distinction between face attribute and vertex attribute is only relevant for `Mesh`. The API should only expose simple `eval_attrbute_X()` methods.